src/frontend/Sider/Journalføring/Standard/validerJournalføring.ts

```diff
@@ -0,0 +1,111 @@
+import { Journalføringsaksjon, JournalføringState } from '../../../hooks/useJournalføringState';
+import { DokumentTitler } from '../../../typer/dokument';
+import { Fagsak } from '../../../typer/fagsak';
+import { JournalpostResponse } from '../../../typer/journalpost';
+import {
+    alleBehandlingerErFerdigstiltEllerSattPåVent,
+    journalføringGjelderKlage,
+} from '../Felles/utils';
+import { Journalføringsårsak } from '../typer/journalføringsårsak';
+
+export const validerJournalføring = (
+    journalResponse: JournalpostResponse,
+    journalpostState: JournalføringState,
+    fagsak: Fagsak
+): string | undefined => {
+    if (journalføringGjelderKlage(journalpostState.journalføringsårsak))
+        return validerKlageJournalføring(journalResponse, journalpostState);
+    return validerStandardJournalføring(journalResponse, journalpostState, fagsak);
+};
+
+const validerKlageJournalføring = (
+    journalResponse: JournalpostResponse,
+    journalpostState: JournalføringState
+): string | undefined => {
+    const valideringsfeil = validerFellesFelter(journalResponse, journalpostState);
+
+    if (valideringsfeil) return valideringsfeil;
+
+    if (
+        journalpostState.journalføringsaksjon === Journalføringsaksjon.OPPRETT_BEHANDLING &&
+        !journalResponse.journalpost.datoMottatt
+    )
+        return 'Mangler gyldig mottatt dato';
+
+    return undefined;
+};
+
+const validerStandardJournalføring = (
+    journalResponse: JournalpostResponse,
+    journalpostState: JournalføringState,
+    fagsak: Fagsak
+): string | undefined => {
+    const valideringsfeil = validerFellesFelter(journalResponse, journalpostState);
+
+    if (valideringsfeil) return valideringsfeil;
+
+    if (journalpostState.journalføringsaksjon === Journalføringsaksjon.OPPRETT_BEHANDLING) {
+        return validerJournalføringTilNyBehandling(journalResponse, journalpostState, fagsak);
+    }
+
+    return undefined;
+};
+
+const validerFellesFelter = (
+    journalResponse: JournalpostResponse,
+    journalpostState: JournalføringState
+): string | undefined => {
+    const { journalføringsårsak, stønadstype, dokumentTitler, nyAvsender } = journalpostState;
+
+    if (journalføringsårsak === Journalføringsårsak.IKKE_VALGT)
+        return 'Mangler journalføringsårsak (Type)';
+
+    if (!stønadstype) return 'Mangler stønadstype';
+
+    if (!harTittelForAlleDokumenter(journalResponse, dokumentTitler))
+        return 'Mangler tittel på et eller flere dokumenter';
+
+    if (journalResponse.journalpost.tema !== 'TSO')
+        return 'Tema på journalføringsoppgaven må endres til «Tilleggsstønader» i Gosys før du kan journalføre dokumentet.';
+
+    if (
+        nyAvsender?.erBruker &&
+        (nyAvsender.navn === undefined ||
+            nyAvsender.navn === '' ||
```

Comment:
Kan bruke `harIkkeVerdi` her
```suggestion
        (harIkkeVerdi(nyAvsender.navn) || ..
```